action/main.py

```diff
@@ -0,0 +1,38 @@
+import os
+import shlex
+import sys
+from pathlib import Path
+from subprocess import PIPE, STDOUT, run
+
+ACTION_PATH = Path(os.environ["GITHUB_ACTION_PATH"])
+ENV_PATH = ACTION_PATH / ".ruff-env"
+ENV_BIN = ENV_PATH / ("Scripts" if sys.platform == "win32" else "bin")
+OPTIONS = os.getenv("INPUT_OPTIONS", default="")
+SRC = os.getenv("INPUT_SRC", default="")
+VERSION = os.getenv("INPUT_VERSION", default="")
+
+run([sys.executable, "-m", "venv", str(ENV_PATH)], check=True)
+
+version_specifier=""
+# TODO: some form of validation for user input VERSION
+if VERSION != "":
+    version_specifier = f"=={VERSION}"
+
+req = f"ruff{version_specifier}"
+pip_proc = run(
+    [str(ENV_BIN / "python"), "-m", "pip", "install", req],
```

Comment:
What about using `pipx install --suffix @ruff-github-action ruff=={version}` here? pipx is [preinstalled](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md) and at least for me locally the entire `pipx intall ruff` is faster than `python -n venv .ruff-env` (1.5s vs. 3.5s)

Reply:
Skipped the pipx install , and just went with a pipx run with version/options as needed